### PR TITLE
add focus functionality to TextField component

### PR DESCRIPTION
**Pull request by ariel7342**
@jonathanadler  here is the focus part, let me know if there is something to fix

### CODE REVIEW

**Comment by ariel7342** [comment]
@jonathanadler pushed a fix for focus ring

**Comment by jonathanadler** [comment]
@ariel7342 
I see that the tests are failing.
I'm not sure the padding works as expected (maybe it's only the line theme?)

File: src/components/TextField/TextField.st.css [hunk 1]
```diff
@@ -165,13 +170,19 @@
...
+
 .suffixWrapper {
     height: 24px;
     display: flex;
     margin: 0 -4px;
     justify-content: flex-end;
     white-space: nowrap;
     vertical-align: middle;
+    padding-left: 12px;
+    padding-right: 12px;
```
Comment (jonathanadler) [inline src/components/TextField/TextField.st.css]: It seems that this is working for the SuccessIcon suffix, but not the ErrorIcon

Comment (ariel7342) [reply src/components/TextField/TextField.st.css]: are you sure? because its the same component with same class just different Icon inside

**Pull request merged**